addons/stock/models/stock_inventory.py

```diff
@@ -353,10 +380,35 @@ class InventoryLine(models.Model):
     theoretical_qty = fields.Float(
         'Theoretical Quantity', compute='_compute_theoretical_qty',
         digits=dp.get_precision('Product Unit of Measure'), readonly=True, store=True)
-    inventory_location_id = fields.Many2one(
-        'stock.location', 'Inventory Location', related='inventory_id.location_id', related_sudo=False, readonly=False)
+    difference_qty = fields.Float('Difference', compute='_compute_difference',
+        help="Indicates the gap between the product's theoretical quantity and its newest quantity.",
+        readonly=True)
+    inventory_date = fields.Datetime('Inventory Date', readonly=True,
+        default=fields.Datetime.now(),
+        help="Last date at which the On Hand Quantity has been computed.")
+    outdated = fields.Boolean(String='Quantity oudated', compute='_compute_outdated',
+        default=False, store=False)
     product_tracking = fields.Selection('Tracking', related='product_id.tracking', readonly=True)
 
+    @api.depends('product_qty', 'theoretical_qty')
+    def _compute_difference(self):
+        for line in self:
+            line.difference_qty = line.product_qty - line.theoretical_qty
+
+    @api.depends('product_id.stock_move_ids')
+    def _compute_outdated(self):
+        for line in self:
+            if not line.outdated:
+                quants = self.env['stock.quant'].search([
+                    ('product_id', '=', line.product_id.id),
+                    ('location_id', '=', line.location_id.id),
+                    ('lot_id', '=', line.prod_lot_id.id),
+                    ('package_id', '=', line.package_id.id)
+                ])
```

Comment:
use _gather? or extract some code from get_inventory_lines_values to use the exact same mechanism?